src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1249,9 +1256,13 @@ impl<T: 'static> EventProcessor<T> {
             .into_iter()
             .filter(|k| *k >= KEYCODE_OFFSET)
         {
-            let scancode = (keycode - KEYCODE_OFFSET) as u32;
-            let keysym = wt.xconn.keycode_to_keysym(keycode);
-            let virtual_keycode = events::keysym_to_element(keysym as c_uint);
+            let keycode = (keycode - KEYCODE_OFFSET) as u32;
+            let mut ker = kb_state.process_key_event(keycode, state);
```

Comment:
What's the reason you are subtracting the `KEYCODE_OFFSET` from `keycode` here? After drilling down into `process_key_event` I found that you are adding 8 back to keycode before using it in `get_one_sym_raw`.

It seems not only unnecessary but also error prone to add/subtract this constant to/from the keycode.

Reply:
The keycodes we get from X11 are at a constant offset from what's defined in the kernel's headers. I can't quite recall if Wayland works the same.

I also can't quite recall why I decided upon the exact places where I add and subtract the offset, but it sure made sense at the time.

Reply:
Ok, so X11 gives us keycodes in the range `8..=255`, while Wayland gives us keycodes starting at `0` (I don't know if there's an upper bound beyond what the integer can store). `libxkbcommon` expects the keycodes in X11's offset-by-eight convention. With that in mind, I guess I'll just do a single `+8` on the Wayland side instead of sprinkling it all over the place.